ena-submission/src/ena_deposition/check_external_visibility.py

```diff
@@ -0,0 +1,399 @@
+"""
+Query ENA and NCBI to check if a given accession is publicly visible
+Add timestamp to project/sample/assembly table when first publicly visible
+1. Find all accessions that don't yet have a publicly visible timestamp
+2. Query ENA and NCBI for these accessions
+3. If publicly visible, update the timestamp in the database
+4. Repeat periodically
+"""
+
+import logging
+import threading
+import time
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from http import HTTPStatus
+from typing import Any
+
+import pytz
+import requests
+from psycopg2.pool import SimpleConnectionPool
+
+from ena_deposition.config import Config
+from ena_deposition.submission_db_helper import (
+    AssemblyTableEntry,
+    ProjectTableEntry,
+    SampleTableEntry,
+    Status,
+    TableName,
+    db_init,
+    find_conditions_in_db,
+    update_db_where_conditions,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class EntityType(Enum):
+    PROJECT = "project"
+    SAMPLE = "sample"
+    ASSEMBLY = "assembly"
+
+
+@dataclass
+class ColumnCheckConfig:
+    """Configuration for checking a specific table column"""
+
+    table_name: TableName
+    entry_class: type
```

Comment:
I think this might be clearer

```suggestion
    entry_type: type
```